OpenRA.Game/Network/Order.cs

```diff
@@ -118,8 +118,10 @@ public static Order Deserialize(World world, BinaryReader r)
 							{
 								case TargetType.Actor:
 								{
-									if (world != null && TryGetActorFromUInt(world, r.ReadUInt32(), out var targetActor))
-										target = Target.FromActor(targetActor);
+									var actorID = r.ReadUInt32();
+									var actorGeneration = r.ReadInt32();
+									if (world != null && TryGetActorFromUInt(world, actorID, out var targetActor))
+										target = Target.FromSerializedActor(targetActor, actorGeneration);
```

Comment:
We create a new `Target` here from the actor with the generation we were told to use, but at this point the actor's generation could have changed. I guess you're relying on the targeting code later dropping the target because of that mismatch, but spitballing here, why not:
Add an (optional) `generation` parameter to `Order.TryGetActorFromUInt()` and maybe even `World.GetActorById()` and pass the generation you just read. If an actor is found but the generation doesn't match, pretend no actor is found and save us some trouble with a target that we know will eventually be invalid.
*(alternatively you can check the generation here after getting the actor and still not create a `Target`, but I like my first suggestion better)*

Reply:
This would take us back to square one wrt the local and remote clients having different target states!

I don't necessarily disagree with the motivation, but if we're going to change this it should be done as part of a larger rewrite instead of adding yet another edge case that could go wrong.

Reply:
Why have different? The serialized order includes the targeted generation and then on the receiving end we make sure to ***only create a `Target` if that is the current generation*** 🤔   Maybe I'm missing something, but would the ***generations*** differ between clients?

Reply:
The sending client would have a target that is internally an actor but returns `TargetType.Invalid` while the receiving client would have a target that is internally `TargetType.Invalid`. We haven't yet proved that there aren't more cases where this distinction matters.

Reply:
Hm ok I guess that makes sense.

Reply:
Hm ok I guess that makes sense.
I figured the sending client would recreate its target based on the incoming order. Did we use to do that and change it as part of the netcode rewrite or am I going crazy? Anyway, that's a valid explanation of what I was missing.

Reply:
> Did we use to do that and change it as part of the netcode rewrite

Yes, 6421c1751546ad275e4e00546a78046761c45ce2 (second point).

Reply:
> Did we use to do that and change it as part of the netcode rewrite or am I going crazy?

Yes we did use to do that because orders would get serialized and deserialized even for the local client, i.e. when they never need to go over the network. That was changed to not serializing and deserializing to save the unnecessary overhead.